ecommerce/views.py

```diff
@@ -152,3 +164,54 @@ def post(self, request, *args, **kwargs):
                     )
         # The response does not matter to CyberSource
         return Response()
+
+
+class CouponsView(ListModelMixin, GenericViewSet):
+    """
+    View for coupons API. This is a read-only API showing the user
+    - what coupons they have available if those coupons would be automatically applied on checkout
+    - what coupons they have for a given coupon code, even if those coupons wouldn't be automatically applied
+    """
+    authentication_classes = (
+        SessionAuthentication,
+        TokenAuthentication,
+    )
+    permission_classes = (IsAuthenticated,)
+
+    def get_queryset(self):
+        """List coupons which a user is allowed to see"""
+        return [
+            coupon for coupon in Coupon.objects.all()
+            if is_coupon_redeemable(coupon, self.request.user) and (
+                coupon.is_automatic or UserCoupon.objects.filter(user=self.request.user, coupon=coupon).exists()
+            )
+        ]
+
+    serializer_class = CouponSerializer
+
+
+class UserCouponsView(APIView):
+    """
+    View for coupon/user attachments. Used to create attachments for a user for a coupon.
+    """
+    permission_classes = (IsLoggedInUser, IsAuthenticated,)
+    authentication_classes = (
+        SessionAuthentication,
+        TokenAuthentication,
+    )
+
+    def post(self, request, *args, **kwargs):
+        """Attach a coupon to a user"""
+        code = self.kwargs['code']
```

Comment:
you can put `code` as an `arg` (or `kwarg` if you need a default value) in the definition of the function and avoid this code